netket/utils/history.py

```diff
@@ -16,30 +16,22 @@
 from numbers import Number
 import jax.numpy as jnp
 
+from plum import dispatch
 
-def _is_scalar(val):
-    """
-    Returns true if the input is a scalar-like object.
+from .numbers import dtype, is_scalar
 
-    Checks whever it is a Python's number, or any numpy-API
-    scalar.
-    """
-    if isinstance(val, Number):
-        return True
-    elif np.isscalar(val):
-        return True
-    elif jnp.isscalar(val):
-        return True
-    elif hasattr(val, "ndim"):
-        if val.ndim == 0:
-            return True
-        else:
-            return False
 
-    return False
+def raise_if_len_not_match(length, expected_length, string):
+    if length != expected_length:
+        raise ValueError(
+            """
+            Length mismatch: expected object of length {expected_length}, but
+            got object of length {length} for key {string}.
+            """
+        )
 
 
-class History:
+class MVHistory:
```

Comment:
I'm very much in favor of merging `History` and `MVHistory`. Maybe we could just call this `History` now, if it's the only class?